storage/local/storage.go

```diff
@@ -360,11 +371,28 @@ func (s *memorySeriesStorage) GetMetricForFingerprint(fp clientmodel.Fingerprint
 
 // AppendSamples implements Storage.
 func (s *memorySeriesStorage) AppendSamples(samples clientmodel.Samples) {
+	if len(samples) == 0 {
```

Comment:
The present state would result in a bug if we removed the check, because of OpenTSDB.
It would also create a lot of lock contention and allocations for nothing.
The initial check is so fast, it's really not appropriate to call it "needless work".